manager/state/raft/raft_test.go

```diff
@@ -595,3 +597,91 @@ func TestRaftUnreachableNode(t *testing.T) {
 	raftutils.CheckValue(t, clockSource, nodes[1], value)
 	raftutils.CheckValue(t, clockSource, nodes[2], value)
 }
+
+func TestStress(t *testing.T) {
+	t.Parallel()
+
+	// Bring up a 5 nodes cluster
+	nodes, clockSource := raftutils.NewRaftCluster(t, tc)
+	raftutils.AddRaftNode(t, clockSource, nodes, tc)
+	raftutils.AddRaftNode(t, clockSource, nodes, tc)
+	defer raftutils.TeardownCluster(t, nodes)
+
+	stop := make(chan struct{})
+	// sync clockSource with real clock
+	go func() {
+		ticker := time.NewTicker(10 * time.Millisecond)
+		defer ticker.Stop()
+		for {
+			select {
+			case <-stop:
+				return
+			case <-ticker.C:
+				clockSource.Increment(10 * time.Millisecond)
+			}
+		}
+	}()
+
+	// number of nodes that are running
+	nup := len(nodes)
+	// record of nodes that are down
+	idleNodes := map[int]struct{}{}
+
+	// record of proposed id and its corresponding value
+	ids := []string{}
+	values := []*api.Node{}
+
+	for iters := 0; iters < 1000; iters++ {
+		// keep proposing new values and killing leader
+		leader := -1
+		for i := 1; i <= 5; i++ {
+			if nodes[uint64(i)] != nil {
+				id := strconv.Itoa(rand.Int())
+				val, err := raftutils.ProposeValue(t, nodes[uint64(i)], id)
+				if err == nil {
+					leader = i
+					ids = append(ids, id)
+					values = append(values, val)
+				}
+			}
+		}
+
+		if (rand.Int() % 1000) < 100 {
+			// sleep a while to wait for potential election
+			time.Sleep(500 * time.Millisecond)
```

Comment:
`Sleep` here is to increase clockSource, and to **accelerate** election progress. If election doesn't finish within 500 ms, test will continue to run and cover the case of `NoLeaderShip`. 

And I think election will finish eventually, at some time.